abnf/odata-abnf-testcases.yaml

```diff
@@ -3684,3 +3685,12 @@ TestCases:
   - Name: ODATA-633 hex pair bug
     Rule: stringInUrl
     Input: '"b%75g"'
+
+  - Name: non-ALPHA characters
+    Rule: odataRelativeUri
+    Input: Customers('Bigfoot')/Schuhgröße
+
+  - Name: encoded non-ALPHA characters
+    Rule: odataRelativeUri
+    FailAt: 28
+    Input: Customers('Bigfoot')/Schuhgr%c3%b6%c3%9fe
```

Comment:
This test case must succeed because the special characters are correctly percent-encoded.

Note: the ABNF header comment says
https://github.com/oasis-tcs/odata-abnf/blob/310e33e18f24f892d3b8f1caf288ffac29f204c2/abnf/odata-abnf-construction-rules.txt#L46-L55

The `ö` and `ß` characters are definitely outside of the "unreserved" set:
https://github.com/oasis-tcs/odata-abnf/blob/310e33e18f24f892d3b8f1caf288ffac29f204c2/abnf/odata-abnf-construction-rules.txt#L1212

Reply:
Reverted your commit.

Reply:
> This test case must succeed because the special characters are correctly percent-encoded.

But it fails after reverting my previous commit:
> encoded non-ALPHA characters fails at 28: Customers('Bigfoot')/Schuhgr%c3%b6%c3%9fe

Reply:
Yes, because the ABNF is incomplete, and this PR aims at pointing this out more clearly:
https://github.com/oasis-tcs/odata-abnf/blob/a0da9e2e33ee3ae5569bcbe7d394b49005e6d9a1/abnf/odata-abnf-construction-rules.txt#L802-L804